File: auth-server/src/test/java/no/vicx/authserver/ActuatorHealthTest.java
package no.vicx.authserver;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.boot.test.web.server.LocalManagementPort;
import org.springframework.http.HttpStatus;
import org.springframework.web.util.UriComponentsBuilder;

import java.net.URI;

import static org.junit.jupiter.api.Assertions.assertEquals;


@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class ActuatorHealthTest {

    @LocalManagementPort
    private int managementPort;

    @Autowired
    TestRestTemplate restTemplate;

    @ParameterizedTest
    @ValueSource(strings = {"liveness", "readiness"})
    @DisplayName("Health Endpoint Probes Should Return HTTP 200")
    void testHealthProbes(String probeName) {
        URI uri = UriComponentsBuilder
                .fromUriString("http://localhost:{port}/actuator/health/{probeName}")
                .buildAndExpand(managementPort, probeName)
                .toUri();

        var response = restTemplate.getForEntity(uri, String.class);

        assertEquals(HttpStatus.OK, response.getStatusCode());
        assertEquals("{\"status\":\"UP\"}", response.getBody());
    }
}

File: auth-server/src/test/kotlin/no/vicx/authserver/ActuatorHealthTest.kt
package no.vicx.authserver

import io.kotest.assertions.assertSoftly
import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.data.Row1
import io.kotest.data.forAll
import io.kotest.matchers.shouldBe
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.boot.test.web.server.LocalManagementPort
import org.springframework.http.HttpStatus
import org.springframework.web.util.UriComponentsBuilder

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class ActuatorHealthTest(
    @LocalManagementPort val managementPort: Int,
    restTemplate: TestRestTemplate,
) : BehaviorSpec({

        Given("an actuator") {

            forAll(
                Row1("liveness"),
                Row1("readiness"),
            ) { probeName ->

                When("calling getForEntity: $probeName") {
                    val uri =
                        UriComponentsBuilder
                            .fromUriString("http://localhost:{port}/actuator/health/{probeName}")
                            .buildAndExpand(managementPort, probeName)
                            .toUri()

                    val response = restTemplate.getForEntity(uri, String::class.java)

                    Then("Health endpoint probes should return OK") {
                        assertSoftly(response) {
                            statusCode shouldBe HttpStatus.OK
                            body shouldBe "{\"status\":\"UP\"}"
                        }
                    }
                }
            }
        }
    })
